llvm/test/Transforms/InstCombine/saturating-add-sub.ll

```diff
@@ -1793,10 +1793,9 @@ define i32 @not_uadd_sat(i32 %x, i32 %y) {
 
 define i32 @not_uadd_sat2(i32 %x, i32 %y) {
 ; CHECK-LABEL: @not_uadd_sat2(
-; CHECK-NEXT:    [[A:%.*]] = add i32 [[X:%.*]], -2
-; CHECK-NEXT:    [[C:%.*]] = icmp ugt i32 [[X]], 1
-; CHECK-NEXT:    [[R:%.*]] = select i1 [[C]], i32 [[A]], i32 -1
-; CHECK-NEXT:    ret i32 [[R]]
+; CHECK-NEXT:    [[X:%.*]] = call i32 @llvm.umax.i32(i32 [[X1:%.*]], i32 1)
```

Comment:
Missing fold: https://alive2.llvm.org/ce/z/LnDhkk
```
define i32 @src(i32 %x, i32 %y) {
  %a = add i32 %x, -2
  %c = icmp ult i32 %x, 2
  %r = select i1 %c, i32 -1, i32 %a
  ret i32 %r
}

define i32 @tgt(i32 %x) {
  %a = call i32 @llvm.umax(i32 %x, i32 1)
  %r = add i32 %a, -2
  ret i32 %r
}
```

Reply:
Thank you for the feedback.

While this fold is similar to the one in this PR, I don't think this PR can fold the provided example because: 

In the example, `C1 = 2`, `C2 = -2`, `C3 = 1` and `BOp = add`.

This PR can fold only when `C3 == BOp C1 C2` and this does not hold true for the example.